gpytorch/lazy/lazy_tensor.py

```diff
@@ -838,7 +839,11 @@ def cat_rows(self, cross_mat, new_mat, generate_roots=True, **root_decomp_kwargs
                 new_inv_root = torch.triangular_solve(Q.transpose(-2, -1), R, upper=True).solution.transpose(-2, -1)
             else:
                 # solve with least squares
-                new_inv_root = torch.lstsq(Q.transpose(-2, -1), R).solution.transpose(-2, -1)
+                if R.ndim > 2:
+                    # temporary hack around PyTorch limitation, this is no good...
```

Comment:
Batching is supported in PyTorch master. This is not appropriately reflected in the PyTorch docs, but is supported per my reading of ATen internals.

The input shapes should satisfy the following criteria:
1. `A` should be of dimension >= 2
2. `b` should be of dimension >= 1
3. the difference between `A.dim` and `b.dim` should be 0 and 1
4. if the difference between `A.dim` and `b.dim` is 1, then `b` is unsqueezed in the last dimension i.e., `b.unsqueeze(-1)`.
5. let the shapes of `A` and `b` after step 4 be `A_shape` and `b_shape`. Then `A_shape = b_shape`.

Reply:
Ah ok so this is supported in `torch.linalg.lstsq`, but not in `torch.lstsq`.

Reply:
Also, it's not available in 1.8, so we'll need to wait until we can bump pytorch to 1.9 for this.